rpc_jobs/jjb_setup.yml

```diff
@@ -8,63 +8,75 @@
+        } else{
+          updateArgs = ''
+        }
+        String sourcesArgs = jobSources.collect {"--job-source ${it.repo};${it.commitish}"}.join(' ')
         common.internal_slave(){
           try {
-            dir("rpc-gating"){
-              stage('Run JJB'){
-                withCredentials([
-                  usernamePassword(
-                    credentialsId: "service_account_jenkins_api_creds",
-                    usernameVariable: "JENKINS_USER",
-                    passwordVariable: "JENKINS_API_PASSWORD"
-                  )
-                ]){
-                  lock('jjb'){
-                  sh """#!/bin/bash -xe
-                    source ../.venv/bin/activate
+            stage('Run JJB'){
+              withCredentials([
+                usernamePassword(
+                  credentialsId: "service_account_jenkins_api_creds",
+                  usernameVariable: "JENKINS_USER",
+                  passwordVariable: "JENKINS_API_PASSWORD"
+                )
+              ]){
+                lock('jjb'){
+                  dir("jenkins_jobs"){
+                    sh """#!/bin/bash -xe
+                      source .venv/bin/activate
 
-                    cat > jenkins_jobs.ini << EOF
-      [job-builder]
-      ignore_cache=False
-      keep_descriptions=False
-      include_path=.:scripts:~/git/
-      recursive=True
-      exclude=.*manual:./development
-      allow_deplicates=False
+                      cat > jenkins_jobs.ini << EOF
+        [job-builder]
+        ignore_cache=False
+        keep_descriptions=False
+        include_path=.:scripts:~/git/
+        recursive=True
+        exclude=.*manual:./development
+        allow_deplicates=False
 
-      [jenkins]
-      url=${JENKINS_URL}
-      EOF
+        [jenkins]
+        url=${JENKINS_URL}
+        EOF
 
-                    if [[ "$RPC_GATING_BRANCH" == "master" ]] && [[ "$JOBS" == "-r rpc_jobs" ]]; then
-                        echo "Setting --delete-old as RPC_GATING_BRANCH=master and JOBS='-r rpc_jobs'"
-                        UPDATE_ARGS="--delete-old"
-                    fi
+                      jjb_paths=\$(rpc-gating/scripts/jjb-path-setup.py '${sourcesArgs}')
 
-                    jenkins-jobs --conf jenkins_jobs.ini \
-                                 --user $JENKINS_USER \
-                                 --password $JENKINS_API_PASSWORD \
-                                 --ignore-cache \
-                                 \$JJB_ARGS update \$UPDATE_ARGS $JOBS
-                  """
-                  } // lock
-                } // with credentials
-              } // stage
-            } // dir
+                      jenkins-jobs --conf jenkins_jobs.ini \
+                                   --user $JENKINS_USER \
+                                   --password $JENKINS_API_PASSWORD \
+                                   --ignore-cache \
+                                 \$JJB_ARGS update ${updateArgs} \$jjb_paths $JOBS
```

Comment:
If the path is a directory, where does `-r` get added?
Also have you tested that this allows files to refer to each other across repos (eg reference defaults from one repo in another) ?

Reply:
I left that out because `recursive=True` is defined in the configuration file however that doesn't appear to do anything so I've added `-r` back in. Defaults can be shared across repositories but as I understand it that only applies to being able to have a global defaults file and a named defaults file.